api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -96,6 +91,19 @@ type KonnectExtensionSpec struct {
 	DataPlaneLabels map[string]DataPlaneLabelValue `json:"dataPlaneLabels,omitempty"`
 }
 
+// KonnectExtensionKonnectSpec holds the konnect-related configuration.
+type KonnectExtensionKonnectSpec struct {
+	// ControlPlane is the configuration for the Konnect Control Plane.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlane KonnectExtensionControlPlane `json:"controlPlane"`
```

Comment:
Given that we'd be changing nesting of the fields, I think that renaming the `ControlPlaneRef` to `Ref` (and a corresponding json tag: `controlPlaneRef` to `ref`) might make sense.

This would allow 

```
spec:
  konnect:
    controlPlane:
      controlPlaneRef:
        type: konnectID
        konnectID: 12345678-1234-1234-1234-123456789abc
```

to become

```
spec:
  konnect:
    controlPlane:
      ref:
        type: konnectID
        konnectID: 12345678-1234-1234-1234-123456789abc
```

WDYT? I do see that this would make this not use the same cp reference field name as other types but I believe it makes sense. I'm open to suggestions on this though.

Reply:
Ok, I've done some further adjustments as discussed offline. PTAL